Fix #15 - Improve developer README for openstudioreleases (#17)
openstudiocoalition/openstudiocoalition.github.io: openstudioreleases/README.md

# OpenStudioCoalition GitHub IO Release Pages

## Install Yarn and package dependencies

Install the `yarn` package manager. Assuming you have installed nodejs and npm, `npm install -g yarn`

Now install the dependencies:

```shell
yarn install
```

## Setup secrets

### Download Firebase config file

> Go to [Test Project](https://console.firebase.google.com/u/2/project/osc-downloads/overview), you can access this with the OSC email.

Reviewer: ```suggestion
Go to [OSC Downloads](https://console.firebase.google.com/u/2/project/osc-downloads/overview), you can access this with the OSC email.
```

https://support.google.com/firebase/answer/7015592?hl=en#zippy=%2Cin-this-article

> Go to the Test Project's Settings, get values from the osc "Config" snippet and place it at `firebase-config.json` next to this readme, changing the format to be a JSON one:

Reviewer: ```suggestion
Go to the OSC Downloads Project's Settings, get values from the osc "Config" snippet and place it at `firebase-config.json` next to this readme, changing the format to be a JSON one:
```

```json
{
  "apiKey": "XXXXX",
  "authDomain": "XXXXX",
  "projectId": "osc-downloads",
  "storageBucket": "XXXXX",
  "messagingSenderId": "XXXXX",
  "appId": "XXXXX",
  "measurementId": "XXXXX"
}
```

Additionally, do `cp .env.example .env`

Then edit it to add the values to each key.

### Google Analytics

> Set the `GA_ID` in the `.env` file. This is the Measurement ID for the OS App website itself (OS App Git - GA4)

Author: @macumber I am not sure at all this is right. https://github.com/openstudiocoalition/openstudiocoalition.github.io/issues/15#issuecomment-2367795599

> Set the `GA_ID` in the `.env` file. This is the Measurement ID for the OS App website itself (OS App Git - GA4)

Reviewer: ```suggestion
Set the `GA_ID` in the `.env` file. This is the Measurement ID for the osc-downloads data stream.
```

The data is stored in the `users` collection.

The script below will get all users from the database and save to a json file.

For it to work, you need the `firebase-service-account.json` with the private key next to this README.md (Ask the maintainers).

To generate one you will need to go to the [Firebase project Service Accounts settings](https://console.firebase.google.com/u/0/project/osc-downloads/settings/serviceaccounts/adminsdk)

> Generate a new private key, and save it to `firebase-service-account.json`. **Note: as soon as you generate a new private key, the old key will stop working, so ask the maintainers to provide it instead**

Author: @macumber please review this carefully... (also, do you have that json file to share with me?)

Reviewer: I think I gave you the  `firebase-service-account.json` earlier, let me know if you still need it

yarn es scripts/syncReleaseAssets.ts
```

> # TODO: explain this

Author: I also don't get the thing about 

> Upload `cors.json` to the bucket.

Where is that cors.json?!

Reviewer: It's at https://github.com/openstudiocoalition/openstudiocoalition.github.io/blob/master/openstudioreleases/config/cors.json

> # TODO: explain this

Reviewer: ```suggestion
# Set development or production
```

> set `.env.prd` for production build

Author: * [ ] TODO: I don't get that one

> set `.env.prd` for production build

Reviewer: ```suggestion
set `NODE_ENV` in the `.env` to either "development" or "production" before making a build
```
